### PR TITLE
Fix the query scope of the model calculated field

### PR DESCRIPTION
**Pull request by goldmedal**
# Description
The available query scope of the calculated field is the columns of the same model.
Consider the SQL below:
```
SELECT orderkey as key, key | '_123' from orders
```

It works in DuckDB but cannot work in Postgres or BigQuery. We should wrap an additional subquery around the base model query to obtain the correct query scope.

### CODE REVIEW

File: wren-tests/src/test/java/io/wren/testing/postgres/TestWrenWithPostgres.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        try {
+            String customer = requireNonNull(getClass().getClassLoader().getResource("tpch/data/customer.parquet")).getPath();
+            duckdbClient.executeDDL("create table db.tpch.customer as select * from '" + customer + "'");
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+        finally {
+            duckdbClient.close();
+        }
```
Comment (grieve54706) [inline wren-tests/src/test/java/io/wren/testing/postgres/TestWrenWithPostgres.java:86]: How about extract a method name `injectTableToPG`?

**Comment by grieve54706** [comment]
Could you add rewritten SQL before and after to show how different it is directly?

**Review by grieve54706 — APPROVED** [review]
LGTM, please use the method to directly show the purpose for the code block.

**Comment by goldmedal** [comment]
> Could you add rewritten SQL before and after to show how different it is?

Given a model `Customer` with `custkey`, `name`, `address` and a calculated field `name_address`.

Before:
```
WITH "Customer" AS (
  SELECT 
    "Customer"."custkey" "custkey", 
    "Customer"."name" "name", 
    "Customer"."address", "address",
    "Customer"."name_address" "name_address" 
  FROM 
    (
      SELECT 
        c_custkey "custkey", 
        c_name "name", 
        c_address "address", 
        concat(name, '_', address) "name_address" 
      FROM 
        "tpch"."tpch"."customer" "Customer"
    ) "Customer"
) 
SELECT 
  name_address 
FROM 
  "Customer"
```

After:
```
  "Customer" AS (
   SELECT
     "Customer"."custkey" "custkey"
   , "Customer"."name" "name"
   , "Customer"."address", "address"
   , "Customer"."name_address" "name_address"
   FROM
     (
      SELECT
        "Customer"."custkey" "custkey"
      , "Customer"."name" "name"
      , "Customer"."address", "address"
      , concat(name, '_', address) "name_address"
      FROM
        (
         SELECT
           c_custkey "custkey"
         , c_name "name"
         , c_address "addrsss"
         FROM
           "tpch"."tpch"."customer" "Customer"
      )  "Customer"
   )  "Customer"
) 
SELECT name_address
FROM
  "Customer"
```

You can see `name_address` is processed in different scope.

**Pull request merged**